sphinxext/icon.py

```diff
@@ -1,121 +1,322 @@
 import re
-from docutils import nodes
-from docutils.parsers.rst import roles
+import fett
+from docutils import nodes, statemachine
+from docutils.parsers.rst import roles, Directive
 from sphinx.util.nodes import split_explicit_title
+from docutils.utils.error_reporting import ErrorString
 
 PAT_STRIP = re.compile(r'-(?:o|(?:alt))$')
+ALL_SUFFIXES = ('-add', '-edit', '-exclamation', '-remove', '-restart')
+HELP_TEMPLATE = fett.Template('''
+.. list-table::
 
-# A pre-approved list of icons. Approval required to add to this list.
+{{ for name in icons }}
+   * - ``{{ name }}``
+     - :{{ role }}:`{{ name }}`
+{{ end }}
+''')
+
+# Each of the below classes describes an icon font. Each must provide:
+# * A name for the docutils icon node
+# * A name for the docutils role
+# * A prefix for the CSS class attached to the output <span>
+# * A set  of icon names
+# Approval required to add to icon lists.
 # Make sure you update the documentation in github.com:mongodb/docs-meta
 # as well when updating.
-ICONS = set([
-    'caret-right',
-    'caret-square-left',
-    'caret-square-right',
-    'check-circle',
-    'check-square',
-    'copy',
-    'edit',
-    'exclamation-circle',
-    'eye',
-    'globe',
-    'lock',
-    'pencil-alt',
-    'save',
-    'star',
-    'sync-alt',
-    'trash',
-    'trash-alt',
-    'users'
-])
-
-# Discouraged, but not prohibited (yet)
-ICONS.add('arrow-right')
-ICONS.add('asterisk')
-ICONS.add('check')
-ICONS.add('ellipsis-h')
-ICONS.add('minus')
-ICONS.add('plus')
-
-ICONSB = set([
-    'windows',
-])
+
+
+class IconsFA5:
+    """FontAwesome 5 Solid icon font"""
+    NODE_NAME = 'IconFA5'
+    ROLE_NAME = 'icon'
+    CSS_PREFIX = 'fas fa-'
+    ICONS = set([
+        'caret-right',
+        'caret-square-left',
+        'caret-square-right',
+        'check-circle',
+        'check-square',
+        'copy',
+        'edit',
+        'exclamation-circle',
+        'eye',
+        'globe',
+        'lock',
+        'pencil-alt',
+        'save',
+        'star',
+        'sync-alt',
+        'trash',
+        'trash-alt',
+        'users'
+    ])
+
+    # Discouraged, but not prohibited (yet)
+    ICONS.add('arrow-right')
+    ICONS.add('asterisk')
+    ICONS.add('check')
+    ICONS.add('ellipsis-h')
+    ICONS.add('minus')
+    ICONS.add('plus')
+
+
+class IconsBrandsFA5:
+    """FontAwesome 5 Brand icon font"""
+    NODE_NAME = 'IconBrandFA5'
+    ROLE_NAME = 'iconb'
+    CSS_PREFIX = 'fab fa-'
+    ICONS = set([
+        'windows',
+    ])
+
+
+class IconsFA4:
+    """FontAwesome 4.7 icon font"""
+    NODE_NAME = 'IconFA4'
+    ROLE_NAME = 'icon-fa4'
+    CSS_PREFIX = 'fa4 fa4-'
+    ICONS = set([
+        'download',
+        'stop',
+        'info-circle',
+        'times-circle',
+        'exclamation-triangle',
+        'info',
+        'trash-o'
+    ])
```

Comment:
The icons that are used per inspection on 20 Feb 2019:
```
fa-book
fa-caret-down
fa-caret-right
fa-circle
fa-enabled-banner-container
fa-exclamation-circle
fa-exclamation-triangle
fa-globe
fa-info-circle
fa-lock
fa-pencil
fa-plus
fa-plus-square
fa-question-circle
fa-refresh
fa-search
fa-trash-o
```

Reply:
Added!

Reply:
@atsansone `fa-enabled-banner-container` doesn't seem to be a thing in FontAwesome. Can you show me where it's used?

Reply:
@atsansone ping